src/redux/books/booksSlice.js

```diff
@@ -1,13 +1,43 @@
 import { createSlice } from "@reduxjs/toolkit";
 
+const initialState = [
+  {
+    "item_id": "item1",
+    "title": "The Great Gatsby",
+    "author": "John Smith",
+    "category": "Fiction"
+  },
+  {
+    "item_id": "item2",
+    "title": "Anna Karenina",
+    "author": "Leo Tolstoy",
+    "category": "Fiction"
+  },
+  {
+    "item_id": "item3",
+    "title": "The Selfish Gene",
+    "author": "Richard Dawkins",
+    "category": "Nonfiction"
+  }
+]
+
+
 const booksSlice = createSlice({
   name: 'books',
-  initialState: [],
+  initialState,
   reducers: {
     addBook: (state, action) => {
-      state.push(action.payload);
+      const book = {
+        "item_id": `item${state.length+1}`,
+        "title": action.payload.title,
+        "author": action.payload.author,
+        "category": "Undefined"
+      }
+      state.push(book);
```

Comment:
For redux to work correctly, the state must be immutable. This means that whenever we update the redux state, we have to create a copy of the whole state and set values to fields we want to change  in the reducer return [...state, book]

Reply:
> Correction

To avoid using `state.push()` and create a mutable state, I used the destructuring and inside this new state I added the new book